tests/legacy/test_read_exceptions.py

```diff
@@ -208,8 +216,11 @@ def test_timeout_stream(self, generate_task, real_x_series_device):
         assert timeout_exception.value.error_code == DAQmxErrors.SAMPLES_NOT_YET_AVAILABLE
         # We should have read a partial dataset.
         number_of_samples_expected = clocks_to_give - samples_to_read
-        assert timeout_exception.value.samps_per_chan_read == number_of_samples_expected
-        # The data that was succesfully read should have been overwritten.
-        assert not any(element == sentinel_value for element in data[:number_of_samples_expected])
-        # The data that wasn't read should have been unmodified.
-        assert all(element == sentinel_value for element in data[number_of_samples_expected:])
+        number_of_samples_read = timeout_exception.value.samps_per_chan_read
+        assert number_of_samples_read == number_of_samples_expected
+
+        # DAQmx overwrites first channel array slices with other channel data as in AB#2420742
+        # Hence resize of the data is needed to extract each data correctly
+        resized_data = numpy.resize(data, (number_of_channels, number_of_samples_read))
+        for i in range(number_of_channels):
+            assert all(i + 1 == round(element) for element in resized_data[i])
```

Comment:
Use NumPy slicing: `assert (resized_data[i, :] == pytest.approx(...)).all()`

Reply:
The pytest.approx returns a ApproxScaler type and is not working in all(), instead the whole expression is evaluated to bool and tries to call all() on top of it. whereas values types like int, float work in the statement.

Reply:
ok, iterating is fine then